skyportal/facility_apis/slack.py

```diff
@@ -0,0 +1,188 @@
+import json
+import requests
+import textwrap
+
+from astropy.coordinates import SkyCoord
+from astropy import units as u
+
+from . import FollowUpAPI
+from baselayer.app.env import load_env
+
+from ..app_utils import get_app_base_url
+from ..utils import http
+
+env, cfg = load_env()
+
+
+SLACK_URL = f"{cfg['slack.expected_url_preamble']}/services"
+
+
+class SLACKRequest:
+
+    """A dictionary structure for SLACK ToO requests."""
+
+    def _build_payload(self, request):
+        """Payload json for SLACK queue requests.
+
+        Parameters
+        ----------
+
+        request: skyportal.models.FollowupRequest
+            The request to add to the queue and the SkyPortal database.
+
+        Returns
+        ----------
+        payload: json
+            payload for requests. payload includes
+              name: object name
+              instrument_request: instrument name
+              ra: right ascension in decimal degrees
+              dec: declination in decimal degrees
+              filters: comma delimited list of filters
+              exposure_time: exposure time requested in seconds
+              exposure_counts: number of exposures requested
+              username: SkyPortal username of requester
```

Comment:
missing `observation_choices`

Reply:
It's returned as filters here, but called observation_choices in the code to be consistent with the other APIs. Down the line this will probably evolve if we have spectrographs using this (which have IFU's, and therefore the more generic term makes sense).